Project is not displayed in the list (at editor start)

<!-- NOTE! The information you specify will be publicly accessible. -->
### Expected behaviour

1. Start editor.
2. I see the list of my projects to choose from.

### Actual behaviour

1. Start editor.
2. A projects are not displayed in the project list, because they have unicode symbols in the path (i mean characters with code over 128).

### Steps to reproduce

<hr/>

<table>
<tr><td>Build time</td><td>2020-08-27T04:44:14.453565</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>96c820fb9fb9c3d6ddec975e5e1dde4762b0228d</td></tr>
<tr><td>Defold engine sha</td><td>dedf1ed10d96c92df6e361f5494531c79af4c1cf</td></tr>
<tr><td>Defold version</td><td>1.2.172</td></tr>
<tr><td>GPU</td><td>AMD Radeon R9 200 Series</td></tr>
<tr><td>GPU Driver</td><td>4.5.13431 Compatibility Profile Context 16.150.2211.0</td></tr>
<tr><td>Java version</td><td>11.0.1+13</td></tr>
<tr><td>OS arch</td><td>amd64</td></tr>
<tr><td>OS name</td><td>Windows 7</td></tr>
<tr><td>OS version</td><td>6.1</td></tr>
<table>

I confirmed the issue on Ubuntu 20.04 (1.2.170). I boils down to problem reading UTF-8 characters from a java configuration file (prefs.xml).  Providing some more info:

#### prefs.xml
```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE map SYSTEM "http://java.sun.com/dtd/preferences.dtd">
<map MAP_XML_VERSION="1.0">
  <entry key="bundle-generate-build-report?" value="[&quot;~#'&quot;,true]"/>
  ...
  <entry key="open-project-directory" value="[&quot;~#'&quot;,&quot;/home/nando/tmp/ΔοκιμήTest&quot;]"/>
  <entry key="recent-project-entries" value="[&quot;^ &quot;,&quot;/home/nando/tmp/dSoundBufferFull/game.project&quot;,&quot;2021-01-14T21:57:08.059198Z&quot;,&quot;/home/nando/tmp/ΔοκιμήTest/game.project&quot;,&quot;2021-01-14T22:08:30.306786Z&quot;]"/>
</map>
  ...
```

#### Exception

```
2021-01-15 00:11:16.758 ERROR default    editor.prefs - {:line 66, :msg "failed to fetch preference for open-project-directory with value [\"~#'\",\"/home/nando/tmp/ΔοκιμήTest\"]"}
java.lang.RuntimeException: com.fasterxml.jackson.core.JsonParseException: Invalid UTF-8 start byte 0x94
 at [Source: java.io.StringBufferInputStream@a596bdd; line: 1, column: 26]
        at com.cognitect.transit.impl.ReaderFactory$ReaderImpl.read(ReaderFactory.java:119)
        at cognitect.transit$read.invokeStatic(transit.clj:296)
        at editor.prefs$read_transit.invokeStatic(prefs.clj:38)
        at editor.prefs$fn__10076.invokeStatic(prefs.clj:62)
        at editor.prefs$fn__10076.invoke(prefs.clj:59)
        at editor.prefs$fn__10058$G__10036__10067.invoke(prefs.clj:42)
        at editor.welcome$last_opened_project_directory.invokeStatic(welcome.clj:134)
        at editor.welcome$show_welcome_dialog_BANG_.invokeStatic(welcome.clj:572)
        at editor.welcome$show_welcome_dialog_BANG_.invoke(welcome.clj:568)
        at editor.welcome$show_welcome_dialog_BANG_.invokeStatic(welcome.clj:570)
        at editor.boot$select_project_from_welcome$fn__89782.invoke(boot.clj:72)
        at clojure.lang.AFn.run(AFn.java:22)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
        at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
        at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.fasterxml.jackson.core.JsonParseException: Invalid UTF-8 start byte 0x94
 at [Source: java.io.StringBufferInputStream@a596bdd; line: 1, column: 26]
        at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1702)
        at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:558)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._reportInvalidInitial(UTF8StreamJsonParser.java:3542)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._reportInvalidChar(UTF8StreamJsonParser.java:3536)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._finishString2(UTF8StreamJsonParser.java:2539)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._finishAndReturnString(UTF8StreamJsonParser.java:2465)
        at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.getText(UTF8StreamJsonParser.java:315)
        at com.cognitect.transit.impl.JsonParser.parseVal(JsonParser.java:60)
        at com.cognitect.transit.impl.JsonParser.parseArray(JsonParser.java:143)
        at com.cognitect.transit.impl.JsonParser.parseVal(JsonParser.java:56)
        at com.cognitect.transit.impl.JsonParser.parse(JsonParser.java:46)
        at com.cognitect.transit.impl.ReaderFactory$ReaderImpl.read(ReaderFactory.java:117)
        ... 18 common frames omitted
```

--- a/editor/src/clj/editor/prefs.clj
+++ b/editor/src/clj/editor/prefs.clj
@@ -36,7 +36,7 @@
      (transit/write writer value)
      (.toString out)))
 
-(defn- read-transit [s]
+(defn- read-transit [^String s]
   (let [stream (ByteArrayInputStream. (.getBytes s StandardCharsets/UTF_8))
         reader (transit/reader stream :json {:handlers read-handlers})]
     (transit/read reader)))

--- a/editor/src/clj/editor/prefs.clj
+++ b/editor/src/clj/editor/prefs.clj
@@ -17,8 +17,9 @@
             [service.log :as log]
             [cognitect.transit :as transit])
   (:import [java.util.prefs Preferences]
-           [java.io ByteArrayOutputStream StringBufferInputStream]
-           [javafx.scene.paint Color]))
+           [java.io ByteArrayOutputStream ByteArrayInputStream StringBufferInputStream]
+           [javafx.scene.paint Color]
+           [java.nio.charset StandardCharsets]))
 
 (set! *warn-on-reflection* true)
 
@@ -36,7 +37,8 @@
      (.toString out)))
 
 (defn- read-transit [s]
-  (let [reader (transit/reader (StringBufferInputStream. s) :json {:handlers read-handlers})]
+  (let [stream (ByteArrayInputStream. (.getBytes s StandardCharsets/UTF_8))
+        reader (transit/reader stream :json {:handlers read-handlers})]
     (transit/read reader)))
 
 (defprotocol PreferenceStore

--- a/editor/test/editor/prefs_test.clj
+++ b/editor/test/editor/prefs_test.clj
@@ -20,9 +20,12 @@
         c (Color/web "#11223344")]
     (prefs/set-prefs p "foo" [1 2 3])
     (prefs/set-prefs p "color" c)
+    (prefs/set-prefs p "utf8stuff" "/foo/τεστ")
     (is (= [1 2 3] (prefs/get-prefs p "foo" nil)))
     (is (= "unknown" (prefs/get-prefs p "___----does-not-exists" "unknown")))
-    (is (= c (prefs/get-prefs p "color" nil)))))
+    (is (= c (prefs/get-prefs p "color" nil)))
+    (is (= "/foo/τεστ" (prefs/get-prefs p "utf8stuff" nil)))
+    ))
 
 (deftest prefs-load-test
   (let [p (prefs/load-prefs "test/resources/test_prefs.json")]

